### ISSUE TITLE
Errno::ENOMEM when dealing with huge files

### ISSUE THREAD

**Issue by gregshutt** [minimagick/minimagick#284]
A user tried uploading a huge .tif file today and I received the following error report:

Errno::ENOMEM: Cannot allocate memory - mogrify

/home/greg/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/open3.rb:193→ spawn
/home/greg/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/open3.rb:193→ popen_run
/home/greg/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/open3.rb:93→ popen3
/home/greg/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/open3.rb:252→ capture3
mini_magick-4.0.2/lib/mini_magick/shell.rb:38→ block (2 levels) in execute
/home/greg/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/timeout.rb:76→ timeout
mini_magick-4.0.2/lib/mini_magick/shell.rb:37→ block in execute
mini_magick-4.0.2/lib/mini_magick/logger.rb:34→ call
mini_magick-4.0.2/lib/mini_magick/logger.rb:34→ block in benchmark
/home/greg/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/benchmark.rb:294→ realtime
mini_magick-4.0.2/lib/mini_magick/logger.rb:34→ benchmark
mini_magick-4.0.2/lib/mini_magick/logger.rb:21→ debug
mini_magick-4.0.2/lib/mini_magick/shell.rb:36→ execute
mini_magick-4.0.2/lib/mini_magick/shell.rb:20→ run
mini_magick-4.0.2/lib/mini_magick/tool.rb:92→ call
mini_magick-4.0.2/lib/mini_magick/tool.rb:53→ new
mini_magick-4.0.2/lib/mini_magick/image.rb:492→ mogrify
mini_magick-4.0.2/lib/mini_magick/image.rb:378→ method_missing
carrierwave-0.10.0/lib/carrierwave/processing/mini_magick.rb:154→ block in resize_to_fit
carrierwave-0.10.0/lib/carrierwave/processing/mini_magick.rb:264→ manipulate!
carrierwave-0.10.0/lib/carrierwave/processing/mini_magick.rb:153→ resize_to_fit
...

This is on a small Linode instance (1GB memory) running Ubuntu 14.04 LTS.

It looks like it's related to the issue explained here: http://adamniedzielski.github.io/blog/2014/02/05/fighting-paperclip-errno-enomem-error/

There's also a StackOverflow discussion on a similar error:
http://stackoverflow.com/questions/8566703/can-not-allocate-memory

Would it be possible to add support for using the posix-spawn gem if it's installed?

**Comment by janko-m** [comment]
Thanks a lot for reporting this, and for linking useful resources, I really appreciate it. Try the newest version, and add the configuration:

```rb
MiniMagick.configure do |config|
  config.shell_api = "posix-spawn"
end
```

**Comment by gregshutt** [comment]
That fixed it! 

Before updating, I reproduced the issue stopping nginx/Passenger, then running 

    $ stress -m 3 --vm-hang 0

to eat up as much available memory as I could. I restarted nginx, uploaded a large file (9MB jpg, nothing special), and confirmed the error. Then I updated to the new version of MiniMagick, added the suggested config change, and repeated the whole process several times. No errors. Thank you for fixing this so quickly.
